test/system/src/test/java/io/pravega/test/system/AbstractReadWriteTest.java

```diff
@@ -552,6 +573,89 @@ private EventWriterConfig buildWriterConfig() {
         return validEvents;
     }
 
+    void createScope(String scopeName, StreamManager streamManager) {
+        Boolean createScopeStatus = streamManager.createScope(scopeName);
+        if (createScopeStatus) {
+            log.info("Successfully created Scope {}", scopeName);
+        }
+    }
+
+    void createKVT(String scopeName, String kvtName, KeyValueTableConfiguration config, URI controllerURIDirect) {
+        log.info("Creating KVT name:{} in scope {}", kvtName, scopeName);
+        Boolean createKvtStatus = KeyValueTableManager.create(controllerURIDirect).createKeyValueTable(scopeName, kvtName, config);
+        if (createKvtStatus) {
+            log.info("Successfully created KVT {}", kvtName);
+        }
+    }
+
+    void startKVPCreate(String scope, String kvtName, KeyValueTableFactory keyValueTableFactory, URI controllerURIDirect) {
+        testInsertUpdateGetKVP(scope, kvtName, keyValueTableFactory, controllerURIDirect);
+    }
+
+    private void testInsertUpdateGetKVP(String scope, String kvtName, KeyValueTableFactory keyValueTableFactory, URI controllerURIDirect) {
+        log.info("Start KVP operation for KVT {} of scope {}", kvtName, scope);
+        // Creating 5 threads to update KVP table entry
+        for (Integer i = 1; i < 6; i++) {
+            String value = convertStringToBinary("Hello World");
+            Integer keyId = i;
+            CompletableFuture.runAsync(() -> {
+                try {
+                    KeyValueTableInfo kvt = new KeyValueTableInfo(scope, kvtName);
+                    keyValueTable = keyValueTableFactory.forKeyValueTable(kvt.getKeyValueTableName(), KEY_SERIALIZER, VALUE_SERIALIZER, KeyValueTableClientConfiguration.builder().build());
+                    log.info("Start creating KVP id:{} in KVT {}", keyId, kvtName);
+                    // Create KVP
+                    Version insertKvp = keyValueTable.put(null, keyId, value).join();
+                    // get KVP and verify the insert value length
+                    Assert.assertEquals("Value length is not matching", value.length(),
+                            keyValueTable.get(null, keyId).join().getValue().length());
+                    // Call API to update kvp table entry
+                    updateKVPEntry(scope, kvtName, keyId, keyValueTableFactory);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }, Executors.newFixedThreadPool(1));
```

Comment:
You are creating multiple executors that in turn are not close when the test completes. This class already has an `executorService`, why don't you use it to execute your IO tasks?

Reply:
Resolved, added executorService instead of `Executors.newFixedThreadPool(1))`